crates/block-producer/src/withdrawal.rs

```diff
@@ -11,149 +13,307 @@ use gw_types::{
     core::ScriptHashType,
     packed::{
         CellDep, CellInput, CellOutput, CustodianLockArgs, DepositionLockArgs, GlobalState,
-        L2Block, RollupAction, RollupActionUnion, Script, ScriptOpt, Uint128,
-        UnlockWithdrawalViaRevert, UnlockWithdrawalWitness, UnlockWithdrawalWitnessUnion,
-        WithdrawalLockArgs, WithdrawalRequest, WitnessArgs,
+        L2Block, RollupAction, RollupActionUnion, Script, UnlockWithdrawalViaRevert,
+        UnlockWithdrawalWitness, UnlockWithdrawalWitnessUnion, WithdrawalLockArgs,
+        WithdrawalRequest, WitnessArgs,
     },
     prelude::*,
 };
+use parking_lot::Mutex;
 
 use std::{
-    collections::HashSet,
+    collections::{HashMap, HashSet},
+    sync::Arc,
     time::{SystemTime, UNIX_EPOCH},
 };
 
-pub struct GeneratedWithdrawals {
-    pub deps: Vec<CellDep>,
-    pub inputs: Vec<InputCellInfo>,
-    pub outputs: Vec<(CellOutput, Bytes)>,
+lazy_static::lazy_static! {
+    static ref CUSTODIAN_TYPE_SCRIPTS: Arc<Mutex<HashMap<[u8; 32], Script>>> = Arc::new(Mutex::new(Default::default()));
```

Comment:
Do not use global static mutable variable, put this into the block-producer structure.

Reply:
Done